File: Python_Books/算法图解/grokking_algorithms_COPY/04_quicksort/javascript/01_loop_sum.js
/**
 * Sums values in array by loop "for"
 * @param {Array} arr Array of numbers
 * @return {number} Sum of the numbers
 */
function sumLoop( arr ) {
    var result = 0;

    for ( var i = 0; i < newArr.length; i++ ) {
        result += newArr[i];
    }

    return result;
}

var arr = [1, 2, 3, 4];

console.log( sumLoop( arr ) ); // 10

File: Python_Books/算法图解/grokking_algorithms_COPY/04_quicksort/c/01_loop_sum.c
#include <stdio.h>

int sum(int *arr, int size) {
	int total = 0;
	for (int i = 0; i < size; i++) {
		total += arr[i];
	}
	return total;
}

int main(void) {
	int arr[4] = { 1,2,3,4 };
	printf("%d", sum(arr, 4));

	return 0;
}